### PR TITLE
Server: fix buggy configstring transmission

### PR DESCRIPTION
**Pull request by slipher**
The bug is described in #1215.

### CODE REVIEW

File: src/engine/server/sv_init.cpp [hunk 1]
```diff
@@ -82,11 +82,47 @@ void SV_SetConfigstring( int index, const char *val )
 	sv.configstringsmodified[ index ] = true;
 }
 
+static void SendConfigStringToClient( int cs, client_t *cl )
+{
+	char buf[ 1024 ]; // escaped characters, in a quoted context
```
[the remark this answers is not in the document]
Comment (slipher) [reply src/engine/server/sv_init.cpp:87]: The immediately relevant limit here is 1022 which is hard-coded in SV_SendServerCommand. I added a comment about this. I don't believe any buffer declared with size MAX_STRING_CHARS is involved anywhere in the stack.

[the remark this answers is not in the document]
Comment (DolceTriade) [reply src/engine/server/sv_init.cpp:87]: Ah, I didn't exactly understand the comment, but I guess that means that 1022 + \" + \"?

File: src/engine/server/sv_init.cpp [hunk 2]
```diff
@@ -82,11 +82,49 @@ void SV_SetConfigstring( int index, const char *val )
...
+	// max command size for SV_SendServerCommand is 1022, leave a little overhead for the command
+	char *limit = buf + 990;
```
Comment (DolceTriade) [inline src/engine/server/sv_init.cpp:89]: FWIW, the cs chunking code uses 1024 - 64 as the size. Does it make sense to be consistent with that?

**Review by DolceTriade — APPROVED** [review]
LGTM, but didn't test. Maybe Kai can test this out?

**Comment by cu-kai** [comment]
This does _seem_ to fix it, however I saw an instance of:
`Warn: CG_AddPlayerWeapon: weapon 0 (<nullptr>) is not registered`

I only saw this once, and could not reproduce it.

File: src/engine/server/sv_init.cpp (hunk 2 above)
Comment (slipher) [reply src/engine/server/sv_init.cpp:89]: No

**Comment by DolceTriade** [comment]
> Warn: CG_AddPlayerWeapon: weapon 0 (<nullptr>) is not registered

I saw this on my server too but only when I played the td map. I'm unsure if it is related to this issue.

**Comment by cu-kai** [comment]
Yes, also on `td` only on bunker. Haven't seen it since.

**Comment by VReaperV** [comment]
I've seen either this or very similar message before (not on `td`), I think it was on `mission-control`, way before this change.

File: src/engine/server/sv_init.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (slipher) [reply src/engine/server/sv_init.cpp:87]: Well it really only needs to be 2 or 3 char longer than `limit`.

**Pull request merged**